client/src/api/v3/boost.ts

```diff
@@ -1,28 +1,121 @@
 import { emit } from 'api/common/socket';
-import { BoostConfigType } from 'types/boost';
+import {
+  FileConfigT,
+  ConfigBundleT,
+  ConfigObjT,
+  ConfigT,
+  fileConfigTypeToRuntype,
+} from 'types/boost';
+import toast from 'react-hot-toast';
+import { Runtype } from 'runtypes';
+
+type payloadAction = 'upload' | 'delete';
+
+/**
+ * Check that the given content is of the type given
+ *
+ * @param content the content to check for it's type
+ * @param typeRequired the type the content is expected to be
+ *
+ * @returns true if the content satisfies the given type else false
+ */
+function isCorrectContentType(content: any, typeRequired: Runtype) {
```

Comment:
Good point, it was used more than once before all the refactoring. However, I'll add it as a function inside `uploadConfig` but it would still be somewhat 'abstracted' since I feel using a `try` and `catch` statement is best done as a separate function as it removes all the indents you have to include when putting all the code inside `try`, if that makes sense.

Reply:
It's the same amount of indenting as later it's being used as a check for an if statement. I feel this function adds complexity without much value. Would like to know what @hallgchris thinks as this might be just my coding preference.